Makefile

```diff
@@ -18,3 +18,57 @@ gen-config:
 .PHONY: check-config-diff
 check-config-diff: gen-config
 	git diff --exit-code config.sample.yaml README.md docs/03-all-configuration-options.md
+
+.PHONY: clean
+clean:
+	rm -rf client/dist
+	rm -rf .bin
+
+client/node_modules:
+	cd client && npm install
+
+client/dist: client/node_modules
+	cd client && ./build.sh
+
+.PHONY: resources
+resources: client/node_modules client/dist
+
+.PHONY: .bin/linux-amd64/traefik-forward-auth
+.bin/linux-amd64/traefik-forward-auth: resources
+	echo -e "\n###\nBuilding for amd64\n"
+	mkdir -p .bin/linux-amd64
+	CGO_ENABLED=0 \
+	GOOS=linux \
+	GOARCH=amd64 \
+	go build \
+	-o .bin/linux-amd64/traefik-forward-auth \
+	-trimpath \
+	./cmd/traefik-forward-auth
+
+.PHONY: .bin/linux-arm/traefik-forward-auth
+.bin/linux-arm/traefik-forward-auth: resources
+	echo -e "\n###\nBuilding for armhf\n"
+	mkdir -p .bin/linux-arm
+	CGO_ENABLED=0 \
+	GOOS=linux \
+	GOARCH=arm \
+	go build \
+	-o .bin/linux-arm/traefik-forward-auth \
+	-trimpath \
+	./cmd/traefik-forward-auth
+
+.PHONY: .bin/linux-arm64/traefik-forward-auth
+.bin/linux-arm64/traefik-forward-auth: resources
+	echo -e "\n###\nBuilding for arm64\n"
+	mkdir -p .bin/linux-arm64
+	CGO_ENABLED=0 \
+	GOOS=linux \
+	GOARCH=arm64 \
+	go build \
+	-o .bin/linux-arm64/traefik-forward-auth \
+	-trimpath \
+	./cmd/traefik-forward-auth
```

Comment:
Since these are not production builds (which can only be done on the CI), consider having a single `build` that builds for the current architecture. If another arch is needed, it can always be passed as env vars during the `make` invocation.

In essence, have a single `build` step that does `go build` but doesn't have any `GOOS` or `GOARCH` passed

Reply:
The reason why I made it like that is double
* In the end, the github actions could also call the makefile. It would put the commands on a single place, but then the makefile needs some more love.
* You don't always need the same architecture as you're running on. In my case I deploy/test the images on a remote K8s instance. It happens to me that both my machine as the k8s instance are the same, but I often hear people on macbooks needing to build linux-x64 binaries.

Like this the "default" is to build everything, but you can build a single one if you want. So gives the freedom.

Reply:
> In the end, the github actions could also call the makefile

The current command you are proposing is different from the GH actions one. It is missing setting ldflags, which require some work.

> You don't always need the same architecture as you're running on. 

Agreed. I have suggested passing `GOOS` and `GOARCH` when running `make` as env vars in your shell for these scenarios.